api/controllers/AuthController.js

```diff
@@ -856,23 +950,25 @@ module.exports = {
             security: true,
             fail: true,
             oauth: {
+              client_id: clientId,
               code,
             },
           },
         );
+
         // OAuth2 standard error.
         const error = Boom.badRequest('invalid authorization code');
         error.output.payload.error = 'invalid_grant';
-        throw error;
+        return error;
       } else {
         logger.info(
           '[AuthController->accessTokenOauth2] Successful access token request',
           {
             request,
             security: true,
             oauth: {
-              client_id: request.payload.client_id,
-              client_secret: `${request.payload.client_secret.slice(0, 3)}...${request.payload.client_secret.slice(-3)}`,
+              client_id: clientId,
+              client_secret: `${clientSecret.slice(0, 3)}...${clientSecret.slice(-3)}`,
```

Comment:
These two lines are the core bugfix. If you look above you see that `clientId` and `clientSecret` can be derived from two sources, per OAuth protocol, and we have docs describing both methods.

Since Drupal used the POST body method (which puts the data in `request.payload`) then all of our Drupal requests for `access_token` were passing through and getting logged just fine.

However if someone coded up a custom OAuth implmementation like in HID-2136, and they chose the Basic Auth header instead of POST body, the log itself was causing a 500 when in reality everything had been successful. My bad.